[MGDSTRM-10719] Add build/push of bundle and index images

MikeEdgar opened the pull request:
Builds on #869 .

MikeEdgar commented on operator/src/main/java/org/bf2/operator/KasFleetShardOperator.java:
```diff
@@ -13,7 +17,23 @@
+        provider = @Provider(
+                name = "bf2 community",
+                url = "https://github.com/bf2fc6cc711aee1a0c2a/kas-fleetshard"),
+        minKubeVersion = "1.21.0",
```
I suspect it's higher than 1.21 by now. We could bump to 1.24 to align with OCP 4.11

MikeEdgar commented:
Coverage will not display properly for the new module until the `sonar.yml` changes are on `main`. I will add a README for the new module, but this is otherwise ready for review. Most of the new files are test resources for the unit test.

shawkins commented on pom.xml:
```diff
@@ -92,6 +92,7 @@
 
     <modules>
         <module>api</module>
+        <module>bundle</module>
```
Should the bundle module always be active?

MikeEdgar replied:
I figured yes for the build and unit tests, but the actual bundle generation will only happen when the `generate-bundle` is active.

shawkins commented on bundle/src/main/java/org/bf2/kas/fleetshard/bundle/BundleAssembler.java:
```diff
@@ -0,0 +1,526 @@
+
+    void copyPriorityClasses() throws IOException {
+        try (KubernetesClient client = new DefaultKubernetesClient()) {
+            client.load(operatorBundle.getEntry("kubernetes.yml")).get().stream()
```
Just a heads up, later versions of fabric8 require using .items() instead of get() for this.

shawkins reviewed (approved):
LGTM

biswassri reviewed (approved):
Lgtm. Just the $USER change that we discussed. Everything else worked fine for me.

kornys commented on bundle/README.md:
```diff
@@ -0,0 +1,46 @@
+  -Dkas.index.image-group=${QUAY_USERNAME} \
+  -Dkas.index.image=kas-fleetshard-operator-index \
+  -Dkas.index.image-tag=latest \
+  -Dkas.index.build-engine=podman
```
we can also use buildah as we use that in jenkins on ocp pods

MikeEdgar commented:
Updated the documentation with suggestions and reverted the changes to systemtests/pom.xml. That will be done instead in a separate PR. I'll merge once the checks are green (aside from the coverage).
